### ISSUE TITLE
Feature Proposal: lock/unlock action buttons

### ISSUE THREAD

**Issue by Jakenberg** [carbonfive/funnel#4]
While this tool is great, there are still several valid reasons for the Swell team to use the "lock-unlock" system in Slack. I wonder if it would be possible to streamline that process by adding a lock/unlock action button at the right of the funnel bar. If this would be a sensible amount of work, I would expect it to work like this:
1. If my PR has a green checkmark, and the lock is available, I can click on the lock action button to reserve the lock. This means that nobody else can merge to development while I have reserved the lock.
2. Once my PR is merged to development (even better if we check that CI passes), the lock is automatically unlocked.
3. Not necessary but maybe a nice to have: messaging to our #dev-status Slack channel.

**Comment by outofambit** [comment]
@Jakenberg what do you mean by the "funnel bar"? is that referring to this UI element?
<img width="719" alt="screen shot 2017-10-18 at 3 20 22 pm" src="https://user-images.githubusercontent.com/964912/31738213-e9c77bd2-b417-11e7-8186-e74da9bf8d8a.png">

**Comment by Jakenberg** [comment]
@outofambit yes that's it! That makes sense. Looking at CircleCI... they've got a "Details" link at the right, which I think could suffice for this: clicking the link executes an action and text is based on the state of the lock. Albeit little lock icons would be nice :)
